OneCommunityGlobal/HighestGoodNetworkApp
# Diya Updated assignment details on add and viewing blue squares

# Description
Update the blue squares functionality to mention whether it is a manually assigned one or assigned by the cronjob that runs everyweek. 
Any updates to the blue squares must be recorded.

## Related PRS (if any):
This fronted PR is related to the backend PR [#1038](https://github.com/OneCommunityGlobal/HGNRest/pull/1038)

## Main changes explained:
Updated the UserProfile model to hold values of 'manuallyAssigned', 'manuallyAssignedBy' and 'editedBy'

## How to test:
1. check into current branch
2. do `npm install` and `...` to run this PR locally
3. Clear site data/cache
4. Log in as admin user
5. Find a user with previously assigned blue squares by the cronjob. Verify that there is no manual assignment label and no edited by label below the summary text area. (You can edit the summary of this and check for the edited by label)
6. Create a new blue square for any user. Viewing the blue square must have a manual assignment label visible.
7. Change the summary and click on the 'Update Button'. Viewing the blue square now must have both the manual assignment and the edited by label below the summary text area with your username.

## Screenshots or videos of changes:
<img width="517" alt="BlueSquares_Manual_Assignment_UIChanges" src="https://github.com/user-attachments/assets/2a5cb208-9f54-47e1-9aee-3dbe37e40fd9">

## Note (Updated):
Please review that the summary text area is focused with the cursor placed in it on selecting the date.
